libraries/helpers.rb

```diff
@@ -24,7 +24,11 @@ module Helpers
       include Chef::DSL::PlatformIntrospection if Chef::VERSION >= '11.0.0'
 
       def chef_server_user
-        Chef::VERSION >= '11.0.0' ? 'chef_server' : 'chef'
+        if Chef::VERSION >= '11.0.0'
+          system('getent passwd opscode') ? 'opscode' : 'chef_server'
+        else
+          'chef'
+        end
```

Comment:
The disconnect here is that this checks that the chef client version is 11.

However, I question whether this is actually necessary with Chef Server 12? It keeps all its directories separate and the `chef-server-ctl` command manages those with its internal cookbooks. This is also true of Chef Server 11..